src/Magnum/Platform/EmscriptenApplication.cpp

```diff
@@ -0,0 +1,454 @@
+
+    #ifdef MAGNUM_TARGET_GLES3
+    /* WebGL 2 */
+    attrs.majorVersion = 2;
+    #elif defined(MAGNUM_TARGET_GLES2)
+    /* WebGL 1 */
+    attrs.minorVersion = 1;
+    #else
+    #error unsupported OpenGL ES version
+    #endif
+
+    /* Resize window and match it to the selected format */
+    double w, h;
+    emscripten_get_element_css_size("#canvas", &w, &h);
+    emscripten_set_canvas_element_size("#canvas", w, h);
+
+    /* Create surface and context */
+    EMSCRIPTEN_WEBGL_CONTEXT_HANDLE context =
+        emscripten_webgl_create_context("#canvas", &attrs);
+    if(!context) {
+        Error{} << "Platform::EmscriptenApplication::tryCreate(): cannot create WebGL context:";
+                /*<< Implementation::glErrorString(glGetError{}); */
+        return false;
+    }
+
+    /* Make the context current */
+    CORRADE_INTERNAL_ASSERT_OUTPUT(
+            emscripten_webgl_make_context_current(context) == EMSCRIPTEN_RESULT_SUCCESS);
+
+    setupCallbacks();
+
+    /* Return true if the initialization succeeds */
+    return _context->tryCreate();
+}
+#endif
+
+Vector2i EmscriptenApplication::windowSize() const {
+    double w, h;
+    emscripten_get_element_css_size("#canvas", &w, &h);
+    return {Int(w), Int(h)};
+}
+
+void EmscriptenApplication::swapBuffers() {
+    emscripten_webgl_commit_frame();
+}
+
+void EmscriptenApplication::setupCallbacks() {
+    auto mousePressedCallback =
+        [](int, const EmscriptenMouseEvent* event, void* userData) -> Int {
+            MouseEvent e{event};
+            reinterpret_cast<EmscriptenApplication*>(userData)->mousePressEvent(e);
+            return e.isAccepted();
+        };
+    emscripten_set_mousedown_callback("#canvas", this, false, mousePressedCallback);
+
+    auto mouseReleaseCallback =
+        [](int, const EmscriptenMouseEvent* event, void* userData) -> Int {
+            MouseEvent e{event};
+            reinterpret_cast<EmscriptenApplication*>(userData)->mouseReleaseEvent(e);
+            return e.isAccepted();
+        };
+    emscripten_set_mouseup_callback("#canvas", this, false, mouseReleaseCallback);
+
+    auto mouseMoveCallback =
+        [](int, const EmscriptenMouseEvent* event, void* userData) -> Int {
+            MouseMoveEvent e{event};
+            reinterpret_cast<EmscriptenApplication*>(userData)->mouseMoveEvent(e);
+            return e.isAccepted();
+        };
+    emscripten_set_mousemove_callback("#canvas", this, false, mouseMoveCallback);
+
+    // TODO: Trying to map touch to mouse opened quite a can of worms:
+    //  1. Tap is already mapped via click by emscripten => mouse press, mouse move, mouse release
+    //  2. Touch down, move, up is not a tap
+    //  3. If single touch, everything is clear, second touch-down could be handled with canceling, but
+    //     would require a manual mouse release event. Or we could continue just following the first touch?
+    //
+    // Map first touch as left mouse, yes or no?
```

Comment:
@mosra This is up for discussion :)

Reply:
I need to introduce proper touch events anyway, trying to shovel them into mouse events is not a good idea. From those, Sdl2App has the pinch event, but (just checked) that one doesn't even work on Emscripten anyway. And as far as I see, none of the examples need the touch events now, only the mouse emulation for tap, right?

Reply:
The mouse interaction example currently does not work on my phone. Clicking does, but up/down and drag do not.

Reply:
Aparently with SDL, the touch drag is supported, but down/up are also not. For both the mouse-interaction and the arealights this leads to suboptimal behaviour in the showcase 😅

Reply:
If that's just phones, then it's okay I would say. If I remember correctly, even in the original Sdl2App the mouse event has some bug in the coordinate calculation so it's not possible to click on buttons etc.